llvm/lib/IR/DataLayout.cpp

```diff
@@ -233,6 +231,15 @@ DataLayout &DataLayout::operator=(const DataLayout &Other) {
 
 bool DataLayout::operator==(const DataLayout &Other) const {
   // NOTE: StringRepresentation might differ, it is not canonicalized.
+  auto IsPointerSpecEqual = [this, &Other](const PointerSpec &A,
```

Comment:
Couldn't we just add the extra comparison to the existing operator==?

Reply:
I agree that making them std::string will simplify this. I'll do that.